src/vm/object.cpp

```diff
@@ -604,17 +599,18 @@ VOID Object::ValidateInner(BOOL bDeep, BOOL bVerifyNextHeader, BOOL bVerifySyncB
             bSmallObjectHeapPtr = GCHeapUtilities::GetGCHeap()->IsHeapPointer(this, true);
             if (!bSmallObjectHeapPtr)
                 bLargeObjectHeapPtr = GCHeapUtilities::GetGCHeap()->IsHeapPointer(this);
-                
+
             CHECK_AND_TEAR_DOWN(bSmallObjectHeapPtr || bLargeObjectHeapPtr);
         }
 
         lastTest = 3;
 
         if (bDeep)
         {
-            CHECK_AND_TEAR_DOWN(GetHeader()->Validate(bVerifySyncBlock));
+            PTR_ObjHeader ptrObjHeader = GetHeader();
+            CHECK_AND_TEAR_DOWN(ptrObjHeader && ptrObjHeader->Validate(bVerifySyncBlock));
```

Comment:
This is not necessary. Code above ensures that this is non-null.

Reply:
Removed the check.